actions/restore-snapshot

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# THIS ACTION RELIES HEAVILLY ON THE EXISTENCE OF A RESOURCE
+
+# Stop Jenkins
+service jenkins stop
+
+RESOURCE_FILE=$(resource-get workspace)
+RESOURCE_FINGERPRINT=$(sha256sum $RESOURCE_FILE)
+ATOMIC_OPERATION=$(action-get atomic)
+
+if [ ${ATOMIC_OPERATION} != 'False' ]
+then
+    rm -rf /var/lib/jenkins/jobs/*
+    action-set workspace.cleaned="True"
+fi
+
+action-set workspace.sum="$RESOURCE_FINGERPRINT"
+
+# Remove any stale files that may be here
+if [ -f '/var/lib/jenkins/jobs/workspaces.tgz']; then
+    rm /var/lib/jenkins/jobs/workspaces.tgz
+fi
+
+
+
+# Copy the file from the fetched resoure path to the Jenkins Job directory
+# and unpack the thin-backup. This will not remove jobs that dont exist
+# in the tarball.
+cp $RESOURCE_FILE /var/lib/jenkins/jobs/workspaces.tgz
+cd /var/lib/jenkins/jobs
+tar xvfz workspaces.tgz
```

Comment:
We should delete the workspaces.tgz file after expanding it. We still have a copy in $RESOURCE_FILE path.